include/deal.II/sundials/sunlinsol_wrapper.h

```diff
@@ -59,7 +59,9 @@ namespace SUNDIALS
      * @param a_times_fn A function pointer to the function that computes A*v
      * @param linsol_ctx The context object used to set up the linear solver and all vectors
      */
-    SundialsOperator(void *A_data, ATimesFn a_times_fn, SUNContext linsol_ctx);
+    SundialsOperator(void       *A_data,
+                     SUNATimesFn a_times_fn,
+                     SUNContext  linsol_ctx);
```

Comment:
`SUNATimesFn` among the other new types was introduced with version 6.0.0 of SUNDIALS. The old types were deprecated at the same time and scheduled for removal in 7.0.0 (see [changes in v6.0.0](https://sundials.readthedocs.io/en/latest/kinsol/Introduction_link.html#changes-in-v6-0-0))

Since these new names only appear guarded by preprocessor conditionals for the corresponding version, this will be fine.